addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/EnvFileScanRule.java

```diff
@@ -38,8 +49,31 @@ public int getId() {
 
     @Override
     public boolean isFalsePositive(HttpMessage msg) {
+        // We need the response header and the body
+        HttpResponseHeader responseHeader = msg.getResponseHeader();
+        String contentType = responseHeader.getNormalisedContentTypeValue();
         String responseBody = msg.getResponseBody().toString();
-        // It likely is a FP if the response contains neither a comment nor assignment
-        return !responseBody.contains("#") && !responseBody.contains("=");
+
+        // If we don't get a response header, the status code isn't 200 or content length is > 500,
+        // prolly not a .env
+        if (responseHeader.isEmpty()
+                || !HttpStatusCode.isSuccess(200)
```

Comment:
Dont you mean `responseHeader.getStatusCode()` instead of the 200?
200 is always a success code.
The fact this wasnt picked up by tests implies there should be a test for it :)

Reply:
I changed this to your suggestion, hopefully this is correct now (I don't know if I need to check for 200s here or if this rule will only ever trigger on success codes....)